### ISSUE TITLE
Umbrella issue: Package ignite in popular formats

### ISSUE THREAD

**Issue by stealthybox** [weaveworks/ignite#410]
Let's discuss our favorite ways to package linux programs.
If you want to work on one of these or your own favorite format, make an issue for it!

Note that if your package chooses to depend on docker or containerd, you should consider accommodating both the distro and docker upstream packages.
> On ubuntu, this is `docker.io`+`containerd` and `docker-ce`+`containerd.io`

It might be easier to just not depend on a container runtime to avoid breaking user's installations.
_____________
Off the top of my head, here are some package formats we could support:
- debian apt:
  @morph027 began work on this using fpm -- thanks!
  > "btw...me being lazy has created apt repos for ignite/footloose to get the latest updates without curl'ing all the time:"
https://gitlab.com/packaging/ignite/
https://gitlab.com/packaging/footloose
- rpm
- docker
  - using host container runtime
  - with nested container runtime
- pacman
- apk
- nix
- snap
- gofish
- asdf

Mac and Windows builds and packages would be dependent on #33

**Comment by morph027** [comment]
Wohoo, nice it's popping up here ;)

[fpm](https://github.com/jordansissel/fpm) supports a whole lot of those anways, and thanks to golang we are basically just "zipping" a single binary with some metadata, it should not be too complicated. I can take care of this part if you like.

**Comment by morph027** [comment]
> Note that if your package chooses to depend on docker or containerd, you should consider accommodating both the distro and docker upstream packages.
> 
> > On ubuntu, this is `docker.io`+`containerd` and `docker-ce`+`containerd.io`
> 
> It might be easier to just not depend on a container runtime to avoid breaking user's installations.

It should be possible to specify `docker.io|docker-ce` and `containerd|containerd.io`.

Also all other [dependencies](https://github.com/weaveworks/ignite/blob/master/docs/dependencies.md) should be added to the packages....

**Comment by morph027** [comment]
[CNI-Plugins](https://gitlab.com/packaging/cni-plugins) as packages too ;)

**Comment by chanwit** [comment]
@stealthybox could we split this into sub issues for Hacktoberfest?
